File: hacks/Snippets/sd_notify.py
import os, socket

def notify(*args):
    path = os.environ.get("NOTIFY_SOCKET", None)

    if not path:
        return
    elif path[0] == "@":
        path = "\0" + path[1:]

    sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)

    msg = "\n".join(args).encode("utf-8")
    sock.sendto(msg, path)

File: Snippets/sd_notify.pl
#!/usr/bin/env perl
use IO::Socket::UNIX;

sub notify {
	my @args = @_;

	my $path = $ENV{NOTIFY_SOCKET} // return;
	$path =~ s/^@/\0/;

	my $sock = IO::Socket::UNIX->new(Type => SOCK_DGRAM,
					Peer => $path);

	$sock->print(join("\n", @args));
}
